### DISCUSSION TITLE
When are you allowed to enable an extension by default?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#147] [Help]
My question is: When are you allowed to enable an extension by default?
I mean, it's quite obvious that a dark theme shouldn't be enabled by default, but what about a little switch in the header bar or in the dropdown that will turn the dark mode on? Could that be on by default?

**Comment by mxmou** [comment]
Currently only the message count and the load more fix are enabled by default.

**Comment by NoobTracker** [comment]
Yep, I know. But what are the criteria for becoming auto-enabled?

**Comment by Hans5958** [comment]
This is a good question. Some extensions like Refined GitHub and Reddit Enhancement Suite have their "addons" turned on by default, so I am interested to know if SA have rules similar to those extensions, or there are exceptions (such as addons with external requests), or it is allowed at all.

**Comment by NoobTracker** [comment]
Imo extensions are allowed to be on by default when they do stuff like just adding a small button (like the image uploader) but something like a dark theme that changes more or less everything isn't. But that's just my opinion.

**Comment by Hans5958** [comment]
Yeah, something like user styles is a choice of each users, but the others.

**Comment by NoobTracker** [comment]
Okay @WorldLanguages what do you think? So far most userscripts, also those which only change tiny things, are disabled by default.

**Comment by WorldLanguages** [comment]
Well, whether an addon is enabled by default is more or less common sense.
Addons like Scratch Messaging and "Message count in extension icon" will be enabled by default if Scratch Addons replaces Scratch Messaging Extension by griffpatch, for obvious reasons.
Bug fixes like "load more fix" should obviously be enabled by default.
Addons that add obviously missing functionality on Scratch, like adding buttons to uncurate a studio, promote by username, etc. should probably also be enabled by default.
Other addons should not be enabled by default.

**Comment by jeffalo** [comment]
something else to note is that some people might think the extension does nothing because nothing visible is enabled by default. i guess that would be the job of #72 to make sure its clear what addons are/do

**Comment by NoobTracker** [comment]
But promoting by username would be enabled by default.

A question: Does the lack of uploading images to scratch count as adding obviously missing functionality on Scratch?

**Comment by WorldLanguages** [comment]
Do you mean uploading to Scratch forums?

**Comment by NoobTracker** [comment]
Yep

**Comment by WorldLanguages** [comment]
Well - kinda. You can use cubeupload. Uploading directly from the forums in a missing feature, but there's a workaround that isn't very complex to do - unlike leaving a studio, which requires you to use the browser console.

**Comment by NoobTracker** [comment]
But could Jeffalo's image uploader be enabled by default?
And using cubeupload isn't that hard, but it takes like 30s instead of 5, pretty much like leaving a studio.

**Comment by WorldLanguages** [comment]
> But could Jeffalo's image uploader be enabled by default?

At the moment, I don't think so.

> And using cubeupload isn't that hard, but it takes like 30s instead of 5, pretty much like leaving a studio.

Not really: in some studios, you literally can't scroll past the first page of curators, which makes it **impossible** to leave normally.

**Comment by NoobTracker** [comment]
Okay.
Could it be possible to make a list of small extensions like the discuss button and the image uploader that are _recommended_ so you can enable them all at once?

**Comment by Hans5958** [comment]
Dropping this from #157 because it is related to the topic.

> I don't think it's a good idea to enable so many addons to previous Scratch Messaging Extension users, specially considering that extension currently makes no changes to the Scratch website.

I guess those users should be prepared for the migration before we really do the migration. I don't know, if we have lots of users after the migration, we can start turning on things by default.

Tagging #108 because it is also related, probably.

*PS. Is it just too early to close this issue?*

**Comment by NoobTracker** [comment]
I think that it can be closed. 
Just tagging #172 because that's related too

**Comment by NoobTracker** [comment]
Um is `onClick` also called when you click on a button with the mouse wheel?
